src/main/java/de/tum/in/www1/artemis/service/CourseScoreCalculationService.java

```diff
@@ -75,6 +81,26 @@ public CourseScoresDTO calculateCourseScores(long courseId, Collection<Long> stu
                 }
             }
         }
+
+        return new MaxAndReachablePoints(maxPointsInCourse, reachableMaxPointsInCourse);
+    }
+
+    /**
+     * Prepares all entities required for calculateCourseScoresForStudent and calls it to retrieve the total scores for each specified student in the specified course.
+     * If there is a single student id in studentIds, the student id will be filtered in the database as an optimization wherever possible.
+     *
+     * @param courseId   the id of the course to calculate the total scores for.
+     * @param studentIds the id of the students whose scores in the course will be calculated.
+     * @return the max and reachable max points for the given course and the student scores with related plagiarism verdicts for the given student ids.
+     */
+    public CourseScoresForExamBonusSourceDTO calculateCourseScoresForExamBonusSource(long courseId, Collection<Long> studentIds) {
+        Set<Exercise> courseExercises = exerciseRepository.findByCourseIdWithCategories(courseId);
```

Comment:
Seems like the categories are not needed for this method.
```suggestion
        Set<Exercise> courseExercises = exerciseRepository.findByCourseId(courseId);
```